libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/data/MapboxNavigationViewportDataSourceOptions.kt

```diff
@@ -305,4 +312,32 @@ class OverviewFrameOptions internal constructor() {
      * Defaults to `true`.
      */
     var paddingUpdatesAllowed = true
+
+    /**
+     * Options that impact the simplification of geometries framed for overview.
+     *
+     * Simplifying geometries, especially for longer routes, can have a significant impact on the performance of generating frames and each [MapboxNavigationViewportDataSource.evaluate] calls.
+     */
+    class GeometrySimplification internal constructor() {
```

Comment:
should it be more complex, like providing own implementation of simplification. May be just `factor` is not fully successfy all customers  
```kotlin
class GeometrySimplification internal constructor() { 
  setSimplification(simplificationFactor: Int)
  setSimplification(simplificationFunc: (List<List<Point>>) -> List<List<Point>>) // can be interface instead lambda
}

Reply:
That's a good idea! I tried adding the function as a setting but it would have to expose the internal `List<List<List<Point>>>` format of describing the route points. I'm not extremely confident that we won't change it in the future, so I wouldn't like to tie ourselves with it right away. If there's a request for this feature we can certainly add it though since it doesn't change the public API surface.